### DISCUSSION TITLE
[User feedback]

### DISCUSSION THREAD

**Discussion by shirleydst** [keycloak/keycloak#15523] [Archive]
Having attempted to upgrade to keycloak 20.0.1, we have had issues when storing realms and clients in a postgres backend database, but forwarding user-authentication and authorization requests via the userstorageprovider to the existing ad / ldap. Please can you suggest the appropriate configuration? It seems like all authentication requests are being handled by the local storage and returning a user_not_found error, rather than querying the userstorageprovider

**Comment by hmlnarik** [comment]
Thank you for the feedback!

The support for user storage providers in the new store is replaced with more generic approach of custom map store implementation which will not be limited to users area only, but will support all other areas (e.g. roles, clients) in the future as well.

At this moment, it is possible to provide an implementation of a `MapStorage` that takes full control of a given area, e.g. users. It is not yet possible to glue several implementations, this awaits tree storage implementation, i.e. #9742.

There are no plans for supporting the legacy user providers within the new storage at this moment.

**Comment by shirleydst** [comment]
Hi Hynek,

Has Tree Storage been implemented in v22, or is it on a future RoadMap? We would be grateful for an update.

Many thanks
Shirley


________________________________

Capgemini is a trading name used by the Capgemini Group of companies which includes Capgemini UK plc, a company registered in England and Wales (number 943935) whose registered office is at No. 1, Forge End, Woking, Surrey, GU21 6DB.
This message contains information that may be privileged or confidential and is the property of the Capgemini Group. It is intended only for the person to whom it is addressed. If you are not the intended recipient, you are not authorized to read, print, retain, copy, disseminate, distribute, or use this message or any part thereof. If you receive this message in error, please notify the sender immediately and delete all copies of this message.
